pkg/cmd/server/origin/handlers.go

```
@@ -84,6 +90,48 @@ func (c *MasterConfig) authorizationFilter(handler http.Handler) http.Handler {
 	})
 }
 
+// forbidden renders a simple forbidden error
+func forbidden(reason string, attributes authorizer.AuthorizationAttributes, w http.ResponseWriter, req *http.Request) {
+	kind := ""
+	name := ""
+	apiVersion := klatest.Version
+	// the attributes can be empty for two basic reasons:
+	// 1. malformed API request
+	// 2. not an API request at all
+	// In these cases, just assume default that will work better than nothing
+	if attributes != nil {
+		apiVersion = attributes.GetAPIVersion()
+		kind = attributes.GetResource()
+		name = attributes.GetResourceName()
+	}
+
+	// Reason is an opaque string that describes why access is allowed or forbidden (forbidden by the time we reach here).
+	// We don't have direct access to kind or name (not that those apply either in the general case)
+	// We create a NewForbidden to stay close the API, but then we override the message to get a serialization
+	// that makes sense when a human reads it.
+	forbiddenError, _ := kapierrors.NewForbidden(kind, name, errors.New("") /*discarded*/).(*kapierrors.StatusError)
+	forbiddenError.ErrStatus.Message = reason
+
+	// Not all API versions in valid API requests will have a matching codec in kubernetes.  If we can't find one,
+	// just default to the latest kube codec.
+	codec := klatest.Codec
+	if requestedCodec, err := klatest.InterfacesFor(apiVersion); err == nil {
+		codec = requestedCodec
+	}
+
+	formatted := &bytes.Buffer{}
+	output, err := codec.Encode(&forbiddenError.ErrStatus)
+	if err != nil {
+		fmt.Fprintf(formatted, "%s", forbiddenError.Error())
+	} else {
+		_ = json.Indent(formatted, output, "", "  ")
```

If you're going to eat the error, might as well simplify to just:

```go
json.Indent(formatted, output, "", "  ")
```